docs/concepts/programming-model.md

```diff
@@ -33,53 +33,52 @@ Flink offers different levels of abstraction to develop streaming/batch applicat
+    for certain operations. The *DataSet API* offers additional primitives on bounded data sets, like loops or iterations.
 
   - The **Table API** is a declarative DSL centered around *tables*, which may be dynamically changing tables (when representing streams).
-    The [Table API](../dev/table_api.html) follows the (extended) relational model: Tables have a schema attached (similar to tables in relational databases)
+    The [Table API](../dev/table_api.html) follows the (extended) relational model. Tables have a schema attached (similar to tables in relational databases)
     and the API offers comparable operations, such as select, project, join, group-by, aggregate, etc.
-    Table API programs declaratively define *what logical operation should be done* rather than specifying exactly
-   *how the code for the operation looks*. Though the Table API is extensible by various types of user-defined
+    Table API programs declaratively define *what logical operation should to perform* rather than specifying
+   *how the code for the operation looks*. The Table API is extensible by various types of user-defined
     functions, it is less expressive than the *Core APIs*, but more concise to use (less code to write).
-    In addition, Table API programs also go through an optimizer that applies optimization rules before execution.
+    Table API programs also go through an optimizer that applies optimization rules before execution.
 
-    One can seamlessly convert between tables and *DataStream*/*DataSet*, allowing programs to mix *Table API* and with the *DataStream*
+    You can seamlessly convert between tables and *DataStream*/*DataSet*, allowing programs to mix *Table API* and with the *DataStream*
     and *DataSet* APIs.
 
   - The highest level abstraction offered by Flink is **SQL**. This abstraction is similar to the *Table API* both in semantics and
     expressiveness, but represents programs as SQL query expressions.
-    The [SQL](../dev/table_api.html#sql) abstraction closely interacts with the Table API, and SQL queries can be executed over tables defined in the *Table API*.
+    The [SQL](../dev/table_api.html#sql) abstraction closely interacts with the Table API, and you can execute SQL queries over tables defined in the *Table API*.
 
 
 ## Programs and Dataflows
 
-The basic building blocks of Flink programs are **streams** and **transformations**. (Note that the
-DataSets used in Flink's DataSet API are also streams internally -- more about that
-later.) Conceptually a *stream* is a (potentially never-ending) flow of data records, and a *transformation* is an
+The basic building blocks of Flink programs are **streams** and **transformations**. The
+DataSets used in Flink's DataSet API are also streams internally, which this document will cover later. Conceptually a *stream* is a (potentially never-ending) flow of data records, and a *transformation* is an
```

Comment:
Needs line break.

Reply:
@greghogan I'm personally not a fan of hard line breaks, but happy to stick to them, but I'm struggling to figure out what the character limit for the project is, I can't see anything in any style files and it's somewhat varied throughout the docs. If there's a solid number then I'll add to the contributors guide.